### ISSUE TITLE
[8.0.10] Patience = X does not seem to work, model continues to train and overfit

### ISSUE THREAD

**Issue by Denizzje** [ultralytics/ultralytics#560]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

Hello Ultralytics team,

Eager to test out Yolov8 I trained two models, an object detection Yolov5m and an instance segmentation Yolov5l-seg with my own dataset.

Due to it being a highly specialised domain, generally what occured on Yolov5 is that I need about 80-120 epochs to have the model at its best and it starts overfitting way before getting to 300 epochs. Hence I use a patience of 10 or 20 most of the time, in this case I used 10 epochs.

What I noticed with Yolov8 now however, is that the model continues to train while its obvious from the validation results that it is overfitting. YoloV5 would indeed stop the training but YoloV8 seems to continue. Hence I suspect something is not functioning... atleast in the version I was using when I started, and when I finished my trainings I seemed to have been 4 versions behind and we are at 8.0.14 already.

It is most noticiable on the segmentation model, the loss on validation was going up quite alot by the end and mAP dropping steadily. 
![results](https://user-images.githubusercontent.com/25279882/213913128-a71bc6d9-d4ef-424d-b803-13f2d35c8fab.png)

My main concern is that I wonder what epoch best.pt was taken from. There is an inexplicable drop in training loss at the end so I hope that I did not take an overfitting epoch.

### Environment

- Ultralytics YOLOv8.0.10 🚀 Python-3.8.10 torch-1.13.1+cu117 CUDA:0 (NVIDIA A100-SXM4-80GB, 81112MiB)
- Ubuntu 20.04

### Minimal Reproducible Example

I am a silly goose and deleted my instance without copying the exact command used for training. But here is an excerpt from ClearML that the argument Patience=10 was indeed passed to the configuration.

![image](https://user-images.githubusercontent.com/25279882/213913278-acac832d-7f88-4d20-b2a7-56ad7d3b6b42.png)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by AyushExel** [comment]
Ohh yeah Patience has not been implemented yet. We'll fix that soon

**Comment by Denizzje** [comment]
Haha good to know, I will refrain from training for the time being untill its in then 🤣 . 

Any way to see what epoch the best.pt is made from? I wonder if it just grabbed an epoch near the en or something at around epoch 100.

**Comment by AyushExel** [comment]
@Denizzje I'll fix this issue today or tomorrow with the new release

**Comment by glenn-jocher** [comment]
@AyushExel new updates are all ready to go. Should I wait for patience PR from you or push 8.0.15 now?

**Comment by AyushExel** [comment]
@glenn-jocher not patience but there's another small PR coming in a few mins

**Comment by glenn-jocher** [comment]
@Denizzje issue should be resolved in #566 by @AyushExel merged just now and scheduled for 8.0.15 deployment soon. Closing issue, but please let us know if this resolves your problem and if you have any more suggestions or bug report. Thanks!
